webhooks/validating/configuration/src/main.rs

```diff
@@ -789,6 +860,256 @@ mod tests {
+        assert!(resp.allowed);
+    }
+
+    #[test]
+    fn test_validate_configuration_discovery_properties_plain_text() {
+        let discovery_properties = r#"
+        "discoveryProperties": [
+            {
+                "name": "nnnn",
+                "value":"vvvv"
+            }
+        ],
+        "#;
+
+        // plain text discovery properties specified should success
+        let resp = run_validate_configuration_discovery_properties(discovery_properties);
+        assert!(resp.allowed);
+    }
+
+    #[test]
+    fn test_validate_configuration_discovery_properties_plain_text_empty_value() {
+        let discovery_properties = r#"
+        "discoveryProperties": [
+            {
+                "name": "nnnn",
+                "value": ""
+            }
+        ],"#;
+
+        // plain text discovery properties, empty value should success
+        let resp = run_validate_configuration_discovery_properties(discovery_properties);
+        assert!(resp.allowed);
+    }
+
+    #[test]
+    fn test_validate_configuration_discovery_properties_plain_text_no_value() {
+        let discovery_properties = r#"
+        "discoveryProperties": [
+            {
+                "name": "nnnn"
+            }
+        ],"#;
+
+        // plain text discovery properties, value not specified should success
+        let resp = run_validate_configuration_discovery_properties(discovery_properties);
+        assert!(resp.allowed);
+    }
+
+    #[test]
+    fn test_validate_configuration_discovery_properties_plain_text_empty_name() {
+        let discovery_properties = r#"
+        "discoveryProperties": [
+            {
+                "name": ""
+            }
+        ],"#;
+
+        // plain text discovery properties, empty name should success
+        let resp = run_validate_configuration_discovery_properties(discovery_properties);
+        assert!(resp.allowed);
+    }
+
+    #[test]
+    #[should_panic(expected = "Could not parse as Akri Configuration")]
+    fn test_validate_configuration_discovery_properties_plain_text_no_name() {
+        let discovery_properties = r#"
+        "discoveryProperties": [
+            {
+                "value":"vvvv"
+            }
+        ],
+        "#;
+
+        // plain text discovery properties without name should fail, missing field 'name'
+        run_validate_configuration_discovery_properties(discovery_properties);
+    }
+
+    #[test]
+    #[should_panic(expected = "Could not parse as Akri Configuration")]
+    fn test_validate_configuration_discovery_properties_empty_value_from() {
+        let discovery_properties = r#"
+        "discoveryProperties": [
+            {
+                "name": "nnnn",
+                "valueFrom": {
+                }
+            }
+        ],"#;
+
+        // valueFrom discovery properties, not specified should fail, missing content of 'valueFrom'
+        run_validate_configuration_discovery_properties(discovery_properties);
```

Comment:
should there be a `assert!(!resp.allowed);` here?

Reply:
the json is invalid, so #[should_panic(expected = "Could not parse as Akri Configuration")]